test/coverlet.collector.tests/CoverletCoverageDataCollectorTests.cs

```diff
@@ -109,6 +109,37 @@ public void OnSessionEndShouldSendGetCoverageReportToTestPlatform()
             directory.Delete(true);
         }
 
+        [Theory]
+        [InlineData("noValidFormat", 0)]
+        [InlineData("json,cobertura", 2)]
+        [InlineData("json,cobertura,lcov", 3)]
+        public void OnSessionEndShouldSendCoverageReportsForMultipleFormatsToTestPlatform(string formats, int sendReportsCount)
+        {
+            _coverletCoverageDataCollector = new CoverletCoverageCollector(new TestPlatformEqtTrace(), new CoverageWrapper());
+
+            var doc = new XmlDocument();
+            var root = doc.CreateElement("Configuration");
+            var element = doc.CreateElement("Format");
+            element.AppendChild(doc.CreateTextNode(formats));
+            root.AppendChild(element);
+
+            _configurationElement = root;
+
+            _coverletCoverageDataCollector.Initialize(
+                _configurationElement,
+                _mockDataColectionEvents.Object,
+                _mockDataCollectionSink.Object,
+                _mockLogger.Object,
+                _context);
+
+            var sessionStartProperties = new Dictionary<string, object>{{ "TestSources", new List<string> { "Test" }} };
+
+            _mockDataColectionEvents.Raise(x => x.SessionStart += null, new SessionStartEventArgs(sessionStartProperties));
+            _mockDataColectionEvents.Raise(x => x.SessionEnd += null, new SessionEndEventArgs());
+
+            _mockDataCollectionSink.Verify(x => x.SendFileAsync(It.IsAny<FileTransferInformation>()), Times.Exactly(sendReportsCount));
```

Comment:
we could also verify that the expected path are ok checking [FileTransferInformation](https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.testtools.execution.filetransferinformation?view=visualstudiosdk-2017) `Path` property

Reply:
Sorry I can't figure out how I can stub/setup the path for this test. Any idea?

Reply:
Create `var mockDataCollectionSink = new Mock<DataCollectionSink>();` for this test...and mock "SendFileAsync", you can check that all splitted formats are "processed", could work?

Reply:
I'm still struggling here a bit. I tried something like this without success:
```
formats.Split(',').ToList().ForEach(format =>
{
    mockDataCollectionSink.Verify(x => x.SendFileAsync(It.IsAny<DataCollectionContext>(),
        It.Is<string>(y => y.EndsWith($"coverage.{format}")), It.IsAny<bool>()));
});
```

Reply:
```cs
Mock<DataCollectionSink> mockDataCollectionSink = new Mock<DataCollectionSink>();
mockDataCollectionSink.Setup(m => m.SendFileAsync(It.IsAny<FileTransferInformation>())).Callback<FileTransferInformation>(fti =>
{
     // do check on fti.Path
});
```
Check that all file are correctly generated, you can use ` IList<IReporter> reporters = formats.Split(',').Select(f => new ReporterFactory(f).CreateReporter()).ToList();` to generage reporters and read extensions...be sure to handle correctly `sendReportsCount = 0`

Reply:
Ok done. I extended the test.